src/parser/content-evaluator-module.ts

```diff
@@ -172,14 +183,36 @@ export class ContentEvaluatorModule implements Module {
     }
   }
 
-  _generatePrompt(issue: string, comments: { id: number; comment: string }[]) {
+  _generatePrompt(
+    issue: string,
+    allComments: { id: number; comment: string; author: string }[],
+    comments: { id: number; comment: string; author: string }[]
+  ) {
     if (!issue?.length) {
       throw new Error("Issue specification comment is missing or empty");
     }
-    return `I need to evaluate the relevance of GitHub contributors' comments to a specific issue specification. Specifically, I'm interested in how much each comment helps to further define the issue specification or contributes new information or research relevant to the issue. Please provide a float between 0 and 1 to represent the degree of relevance. A score of 1 indicates that the comment is entirely relevant and adds significant value to the issue, whereas a score of 0 indicates no relevance or added value. A stringified JSON is given below that contains the specification and contributors' comments. Each comment in the JSON has a unique ID and comment content. \n\n\`\`\`\n${JSON.stringify(
-      { specification: issue, comments: comments }
-    )}\n\`\`\`\n\n\nTo what degree are each of the comments in the conversation relevant and valuable to further defining the issue specification? Please reply with ONLY a JSON where each key is the comment ID given in JSON above, and the value is a float number between 0 and 1 corresponding to the comment. The float number should represent the degree of relevance and added value of the comment to the issue. The total number of properties in your JSON response should equal exactly ${
-      comments.length
-    }.`;
+
+    return `Instruction: 
```

Comment:
Why do you prefix instruction? 

This is handled via the api you write it as the assistant message. 

Then we can pass in user message after as the "input"

Perhaps that will lead to better results. I've always been mindful of this in my implementations.

Reply:
Adding `Instruction` to the starting of the prompt allows to pass in a common instruction. From all the prompts I have tested, adding this ensures that the LLM model compare across all Comments.